File: documentation/user/en/query/requirements/examples/hierarchy/hierarchy-level-parent.java
final EvitaResponse<SealedEntity> result = session.querySealedEntity(
	query(
		// query hierarchy entity type
		collection("Product"),
		// target "Accessories" category
		filterBy(
			hierarchyWithin(
				"categories",
				attributeEquals("code", "true-wireless")
			)
		),
		require(
			hierarchyOfReference(
				"categories",
				// request computation of all parents
				// until level 2 parent is reached
				parents(
					"parents",
					entityFetch(attributeContent("code")),
					stopAt(level(2))
				)
			)
		)
	)
);

File: documentation/user/en/query/requirements/examples/hierarchy/hierarchy-level-parent.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("code", "true-wireless")
        		)
        	),
        	Require(
        		HierarchyOfReference(
        			"categories",
        			RemoveEmpty,
        			Parents(
        				"parents",
        				EntityFetch(
        					AttributeContent("code")
        				),
        				StopAt(
        					Level(2)
        				)
        			)
        		)
        	)
        )
	)
);